src/Symfony/Component/Config/Resource/GlobResource.php

```diff
@@ -93,29 +97,47 @@ public function getIterator()
             return;
         }
 
-        if (0 !== strpos($this->prefix, 'phar://') && false === strpos($this->pattern, '/**/') && (\defined('GLOB_BRACE') || false === strpos($this->pattern, '{'))) {
-            foreach (glob($this->prefix.$this->pattern, \defined('GLOB_BRACE') ? GLOB_BRACE : 0) as $path) {
-                if ($this->recursive && is_dir($path)) {
-                    $files = iterator_to_array(new \RecursiveIteratorIterator(
-                        new \RecursiveCallbackFilterIterator(
-                            new \RecursiveDirectoryIterator($path, \FilesystemIterator::SKIP_DOTS | \FilesystemIterator::FOLLOW_SYMLINKS),
-                            function (\SplFileInfo $file) { return '.' !== $file->getBasename()[0]; }
-                        ),
-                        \RecursiveIteratorIterator::LEAVES_ONLY
-                    ));
-                    uasort($files, function (\SplFileInfo $a, \SplFileInfo $b) {
-                        return (string) $a > (string) $b ? 1 : -1;
-                    });
-
-                    foreach ($files as $path => $info) {
-                        if ($info->isFile()) {
-                            yield $path => $info;
+            if (0 !== strpos($this->prefix, 'phar://') && false === strpos($this->pattern, '/**/') && (\defined('GLOB_BRACE') || false === strpos($this->pattern, '{'))) {
+                $globBrace = \defined('GLOB_BRACE') ? GLOB_BRACE : 0;
+                $excludedDirs = glob($this->prefix.$this->pattern, $globBrace | GLOB_ONLYDIR);
+                foreach (glob($this->prefix.$this->pattern, $globBrace) as $path) {
+                    if (is_dir($path)) {
+                        if ($this->forExclusion && \in_array($path, $excludedDirs, true)) {
+                            yield $path => new \SplFileInfo($path);
+                            continue;
+                        }
+
+                        if (array_key_exists(str_replace(\DIRECTORY_SEPARATOR, '/', $path), $this->excludedPrefixes)) {
```

Comment:
You can replace these with issets, will be faster. Map of values has been reversed (and `true` used for value) for precisely this reason.

OR we can use `\in_array` which has been optimized into opcode in PHP 7.2 already and is even faster than isset, but you need to reverse the map. It's good opportunity to do this now, because as you added that array into constructor of this class, we won't be able to reverse it later without breaking BC. Obvious con is it will make code slightly slower in PHP 7.1. But IMO worth it, as it's only one PHP version and all future versions will be faster with `in_array`, plus it's more readable (intent is more obvious). WDYT @nicolas-grekas ?

Reply:
in_array cannot be faster than isset, because it still has to do a table scan vs an O(1) hashmap lookup for isset, isn't it?

Reply:
It will reverse it internally, see https://phpinternals.net/articles/optimising_internal_functions_via_new_opcode_instructions#the-in_array-function